src/main/java/Duke.java

```diff
@@ -1,10 +1,342 @@
+import java.util.Scanner;
+
 public class Duke {
+    /**
+     * Words for commands. 
+     */
+    private static final String COMMAND_TODO_WORD = "todo";
+    private static final String COMMAND_DEADLINE_WORD = "deadline";
+    private static final String COMMAND_EVENT_WORD = "event";
+    private static final String COMMAND_MARK_WORD = "done";
+    private static final String COMMAND_LIST_WORD = "list";
+    private static final String COMMAND_EXIT_WORD = "bye";
+    private static final String NO_COMMAND = "";
+    
+    /**
+     * Display messages. 
+     */
+    private static final String MESSAGE_WELCOME = "Hello! I'm Duke, what can I do for you?";
+    private static final String MESSAGE_BORDER = "__________".repeat(6);
+    private static final String MESSAGE_ADDED = "Got it. I've added this task: ";
+    private static final String MESSAGE_MARKED = "Nice! I've marked this task as done: ";
+    private static final String MESSAGE_LIST = "Here are the tasks in your list: ";
+    private static final String MESSAGE_EXIT = "Bye. Hope to see you again soon! ";
+    private static final String MESSAGE_NUMBER_OF_TASKS = "Now you have %s tasks in the list. ";
+    private static final String ERROR_INDEX_OUT_OF_RANGE = "Index out of range. ";
+    private static final String ERROR_NO_COMMAND_RECEIVED = "No command detected. ";
+    private static final String ERROR_INVALID_COMMAND_RECEIVED = "Invalid command. ";
+
+    /**
+     * Filter words to locate date for deadline and event tasks. 
+     */
+    private static final String DEADLINE_DATA_PREFIX_BY = "/by";
+    private static final String EVENT_DATA_PREFIX_AT = "/at";
+
+    /**
+     * List of all tasks. 
+     */
+    private static Task[] tasks;
+    private static String userCommand;
+
+    /**
+     * Maximum number of tasks that can be held. 
+     */
+    private static final int MAX_LIST_SIZE = 100;
+
+    /**
+     * Total number of tasks in task list. 
+     */
+    private static int sizeOfTaskList = 0;
+
+    /**
+     * Scanner for extracting user input. 
+     */
+    private static final Scanner SCANNER = new Scanner(System.in);
+    
+    /**
+     * Gets a task object from the tasks list. 
+     * 
+     * @param index Location of the task on the tasks list. 
+     * @return The task at the index. 
+     */
+    private static Task getTask(int index) {
+        return tasks[index];
+    }
+
+    /**
+     * Gets task number from the input string. 
+     * 
+     * @return Task number. 
+     */
+    private static int getTaskNumber(String command) {
+        int taskNumber = Integer.parseInt(command.substring(command.indexOf(" ") + 1)) - 1;
```

Comment:
There are multiple nested expressions, you can consider breaking it into multiple expressions for better readability. This will also help when handling exceptions from parseInt() in the future

Reply:
Thank you, I will add it to another method instead.